library/x509_crt.c

```diff
@@ -1842,20 +1865,37 @@ static int x509_crt_verifycrl( mbedtls_x509_crt *crt, mbedtls_x509_crt *ca,
             flags |= MBEDTLS_X509_BADCRL_BAD_PK;
 
         md_info = mbedtls_md_info_from_type( crl_list->sig_md );
-        if( mbedtls_md( md_info, crl_list->tbs.p, crl_list->tbs.len, hash ) != 0 )
+        if( ( ret = mbedtls_md( md_info, crl_list->tbs.p,
+                                crl_list->tbs.len, hash ) ) != 0 )
         {
-            /* Note: this can't happen except after an internal error */
+            if( ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )
+            {
+                flags |= MBEDTLS_X509_BADCRL_ALG_NOT_SUPPORTED;
```

Comment:
Because we can't verify the certificate, it is still not trusted, but we still want to know that it was due to an algorithm not supported. As mentioned by @mpg in https://github.com/ARMmbed/mbedtls/pull/2124#pullrequestreview-184622147:
>I particularly like that the new flags are set in addition to the existing NOT_TRUSTED flags rather than replacing them. I think the minimizes the risk of misuse and is an excellent choice.